pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -160,6 +160,10 @@ func (c *Cluster) Start() error {
 		logger.Debugf("storage nodes: %+v", c.Storage.Nodes)
 	}
 	validNodes := k8sutil.GetValidNodes(c.Storage.Nodes, c.context.Clientset, c.placement)
+	// no valid node is ready to run an osd
+	if len(validNodes) == 0 {
+		return fmt.Errorf("no valid node available to run an osd in namespace %s", c.Namespace)
```

Comment:
I'm not sure if returning an actual error is the correct thing to do here, reason being that when the user does not have given a "selector" (placement, nodes list) or a bad one, it isn't really an error of Rook.

I think the better approach would be to log it as a warning/error here and return `nil`. Otherwise I think the Rook operator would go into retry loop to create the Ceph Cluster over and over again with the same "bad" node selector (until the Ceph Cluster CRD has been updated, see next sentence).
The user can then act according to this warning/error log message and update the Ceph Cluster CRD object.